components/NetworkModal.tsx

```diff
@@ -16,8 +17,10 @@ const NetworkModal: React.FC<{
   currentNetwork: string;
 }> = ({ setModalOpen, setModalState, currentNetwork }) => {
   const [open] = useState(true);
+  const { provider } = useWeb3();
```

Comment:
I think, even if we don't add the new provider state..
we can still use api... i.e const { api } = useWeb3();
and then on 

 const changeNetwork = async (selectedNetwork) => {
     if (api && api.isConnected) {
        await api.disconnect();
      }
     updateNetworks(selectedNetwork);
   };